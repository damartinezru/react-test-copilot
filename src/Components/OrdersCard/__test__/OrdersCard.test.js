import OrdersCard from "..";
import { render } from "@testing-library/react";
import '@testing-library/jest-dom/extend-expect';
describe('OrdersCard', () => {
    it('should render the OrdersCard component', () => {
        const { container } = render(<OrdersCard totalPrice={100} totalProducts={3} />);
        expect(container).toMatchSnapshot();
    });
    
    it('should render the OrdersCard component with different props', () => {
        const { container } = render(<OrdersCard totalPrice={200} totalProducts={2} />);
        expect(container).toMatchSnapshot();
    });
    
    it('should render the correct total price', () => {
        const { getByText } = render(<OrdersCard totalPrice={200} totalProducts={2} />);
        expect(getByText('$200')).toBeInTheDocument();
    });

})