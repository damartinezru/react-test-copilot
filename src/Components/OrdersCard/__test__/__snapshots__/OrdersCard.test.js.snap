// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrdersCard should render the OrdersCard component 1`] = `
<div>
  <div
    class="flex justify-between items-center mb-3 border border-black rounded-lg p-4 w-80"
  >
    <div
      class="flex justify-between w-full"
    >
      <p
        class="flex flex-col"
      >
        <span
          class="font-light"
        >
          01.02.23
        </span>
        <span
          class="font-light"
        >
          3
           articles
        </span>
      </p>
      <p
        class="flex items-center gap-2"
      >
        <span
          class="font-medium text-2xl"
        >
          $
          100
        </span>
        <svg
          aria-hidden="true"
          class="h-6 w-6 text-black cursor-pointer"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
            fill-rule="evenodd"
          />
        </svg>
      </p>
    </div>
  </div>
</div>
`;

exports[`OrdersCard should render the OrdersCard component with different props 1`] = `
<div>
  <div
    class="flex justify-between items-center mb-3 border border-black rounded-lg p-4 w-80"
  >
    <div
      class="flex justify-between w-full"
    >
      <p
        class="flex flex-col"
      >
        <span
          class="font-light"
        >
          01.02.23
        </span>
        <span
          class="font-light"
        >
          2
           articles
        </span>
      </p>
      <p
        class="flex items-center gap-2"
      >
        <span
          class="font-medium text-2xl"
        >
          $
          200
        </span>
        <svg
          aria-hidden="true"
          class="h-6 w-6 text-black cursor-pointer"
          fill="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M16.28 11.47a.75.75 0 010 1.06l-7.5 7.5a.75.75 0 01-1.06-1.06L14.69 12 7.72 5.03a.75.75 0 011.06-1.06l7.5 7.5z"
            fill-rule="evenodd"
          />
        </svg>
      </p>
    </div>
  </div>
</div>
`;
